ec/ecresource/deploymentresource/deployment/v2/deployment_update_payload.go

```diff
@@ -148,9 +148,20 @@ func (plan DeploymentTF) updateBasePayloadsForMigration(client *api.API, newDtId
 }
 
 func (plan DeploymentTF) UpdateRequest(ctx context.Context, client *api.API, state DeploymentTF, migrateTemplateRequest *deployments.MigrateDeploymentTemplateOK) (*models.DeploymentUpdateRequest, diag.Diagnostics) {
+	// The alias behaves like this:
+	// - Not set in the config (null/unknown) -> Alias is not managed by terraform and left untouched
+	// - Set to empty string -> Alias will be removed
+	// - Set to non-empty string -> Alias will be set to this value
+	var alias *string
+	if plan.Alias.IsNull() || plan.Alias.IsUnknown() {
```

Comment:
I think a null alias should be treated the same as `alias = ""` (that is the alias is removed). At this point the user has most likely explicitly set the null value in config.

Reply:
Yeah I think that would work. My main concern would be that it starts removing the alias for clusters that have an alias but don't have the field defined.

However I think it won't happen due to the field being computed and the `UseStateForUnknown` modifier. So it will always copy from state, which will always either contain the alias or the empty string.

But I'll test it to see if it really behaves properly.

Reply:
Seems like it won't work: Terraform considers `null` the same as not configuring the field at all (Also documented here: https://developer.hashicorp.com/terraform/language/expressions/types#null)

So even when explicitly setting the config `alias = null` it will end up as an unknown value and the `useStateForUnknownModifier` will set it to the state. 

We could build our own modifier with a different behavior, but since it is documented that `null == not configured` it would then deviate from how terraform usually works.